frontend/src/components/event.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import Image from 'next/image';
+import { Event } from '@/types';
+import 'pure-react-carousel/dist/react-carousel.es.css';
+import Link from 'next/link';
+import Toaster from '@/utils/toaster';
+import { Buildings } from '@phosphor-icons/react';
+import 'pure-react-carousel/dist/react-carousel.es.css';
+import { USER_PROFILE_PIC_URL } from '@/config/routes';
+import postHandler from '@/handlers/post_handler';
+import { SERVER_ERROR } from '@/config/errors';
+import { X } from '@phosphor-icons/react/dist/ssr';
+
+interface Props {
+    event : Event ;
+  setEvents: React.Dispatch<React.SetStateAction<Event[]>>;
+}
+
+const EventComponent = ({ event, setEvents }: Props) => {
+  const removeFlag = () => {
+    const toaster = Toaster.startLoad('Removing Flag', event.id);
+    const URL = `/flags/events/${event.id}`;
+    postHandler(URL, {})
+      .then(res => {
+        if (res.statusCode === 200) {
+          setEvents(prev => prev.filter(p => p.id != event.id));
+          Toaster.stopLoad(toaster, 'Flag Removed', 1);
+        } else {
+          if (res.data.message) Toaster.stopLoad(toaster, res.data.message, 0);
+          else {
+            Toaster.stopLoad(toaster, SERVER_ERROR, 0);
+          }
+        }
+      })
+      .catch(err => {
+        Toaster.stopLoad(toaster, SERVER_ERROR, 0);
+      });
+  };
+  return (
+    <div className="w-full">
```

Comment:
event card should show details about the event. (title, description, created_at, etc)